### DISCUSSION TITLE
How to Handle Merge Conflicts in GitHub?

### DISCUSSION THREAD

**Discussion by ducnhan123-co** [community/community#140191] [Pull Requests] [answered]
Hi everyone,

I'm working on a project with my team, and we recently encountered a merge conflict when trying to merge our branches in GitHub. I would like to understand more about:

What exactly is a merge conflict in GitHub?
What causes these conflicts to happen?
What are the best practices for resolving them?
How can we avoid conflicts in the future?
Any advice, resources, or experiences would be greatly appreciated. Thanks in advance!

**Comment by davevad93** [comment, marked as answer]
Hi @ducnhan123-co ,

Merge conflicts are caused by merging branches that have conflicting commits, so you must decide what changes to incorporate in the final merge.

For more infos, check the docs:

1.  [About merge conflicts](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts)
2. [Resolving a merge conflict on GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github)
3. [Resolving a merge conflict using the command line](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line)

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @ducnhan123-co ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!
